### PR TITLE
Upgrade to rails 4.2.0+ (round 2)

### PR DESCRIPTION
**Pull request by CloCkWeRX**
Just like #964 only more mergable, working.

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -112,5 +112,7 @@ class Application < Rails::Application
     # didn't work for us.
     config.cloudmade_key = '29a2d9e3cb3d429490a8f338b2388b1d'
 
+
```
Comment (pozorvlak) [inline config/application.rb]: Please remove the extra whitespace.

File: config/environments/test.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@
   config.cache_classes = false
 
   # Configure static asset server for tests with Cache-Control for performance
-  config.serve_static_assets = true
+  config.serve_static_files = true
```
Comment (pozorvlak) [inline config/environments/test.rb]: Should we be setting this in `development.rb`?

File: db/migrate/20120903092956_devise_create_users.rb [hunk 3]
```diff
@@ -34,7 +34,7 @@ def change
       # t.string :authentication_token
 
 
-      t.timestamps
+      t.timestamps null: true
```
Comment (pozorvlak) [inline db/migrate/20120903092956_devise_create_users.rb]: Are we currently creating null timestamps? We should probably fix that, if so.

File: config/environments/test.rb (hunk 2 above)
Comment (CloCkWeRX) [inline config/environments/test.rb]: "config.serve_static_files configures Rails itself to serve static files. Defaults to true, but in the production environment is turned off as the server software (e.g. NGINX or Apache) used to run the application should serve static assets instead. "

So, should be ok to leave implicit

File: db/migrate/20120903092956_devise_create_users.rb (hunk 3 above)
Comment (CloCkWeRX) [inline db/migrate/20120903092956_devise_create_users.rb]: Went with what the deprecation message suggested. It's unclear if we will set created_at = Time.now && updated_at = nil.

Comment (pozorvlak) [inline db/migrate/20120903092956_devise_create_users.rb]: OK. Let's go with this for now, and raise an issue to disallow NULL values
for timestamps (may require data migrations).

**Comment by pozorvlak** [comment]
Please fix the whitespace issue, and then this is good to merge.

**Comment by CloCkWeRX** [comment]
Re-running tests

**Pull request merged**